### DISCUSSION TITLE
Enhancement: NekoVM compatibility

### DISCUSSION THREAD

**Discussion by libauror-a** [FunkinCrew/Funkin#1668] [Ideas]
Currently when you try to build with neko, it spits out a error about using a cpp function (relating to DiscordRpc) being used on a Neko target. I'm probally going to make a PR relating to this.

This is what happens happens when you try to build with Neko:

![image](https://user-images.githubusercontent.com/75984453/121233981-84482380-c893-11eb-8182-203bf7e58bae.png)

**Comment by libauror-a** [comment]
This also happens when building with Mac, and linux.

**Comment by ghost** [comment]
Building Linux on Linux machine works fine

**Comment by libauror-a** [comment]
I think building on windows seems to be the problem.

**Comment by ghost** [comment]
> .. I'm probally going to make a PR relating to this.

PLEASE DO, I really don't want to install 4 gigabytes of unwanted space on my PC just to compile on Windows

(you are allowed to react 👎, 😕, or 👀 on this)

**Comment by libauror-a** [comment]
I tried to make DiscordClient CPP only, but when i try to test it with Neko it just hangs, i know that Main.hx isn't the problem because i was testing another game with the Main from the game and using the Neko build target and it worked fine. 

Example (Title bar says Not responding):
![image](https://user-images.githubusercontent.com/75984453/121368332-4999c600-c93b-11eb-9d02-f804f6f1f64f.png)


Edit: it MAY be a problem with main, i changed the initial state to a Hello world! state but it dosen't work.

**Comment by suprstarrd** [comment]
"You cannot access the cpp package while targeting neko" Will look into the `discord_rpc` library, then. For reference, this is the source code for the RPC library - https://github.com/Aidan63/linc_discord-rpc

**Comment by suprstarrd** [comment]
https://github.com/Aidan63/linc_discord-rpc/blob/2d83fa863ef0c1eace5f1cf67c3ac315d1a3a8a5/discord_rpc/DiscordRpc.hx#L3-L7

Well, that explains a lot. Will try building this later and seeing if I can get it to work in Neko (which should be relatively easy!)

**Comment by suprstarrd** [comment]
Update: It will not be easy! Haxe docs show a lot of the CPP functions used aren't supported in Neko, so I'll need to recreate them. That's nice.

**Comment by ghost** [comment]
> Update: It will not be easy! Haxe docs show a lot of the CPP functions used aren't supported in Neko, so I'll need to recreate them. That's nice.

just make a whole frickin engine on it

**Comment by libauror-a** [comment]
![image](https://user-images.githubusercontent.com/75984453/126480818-5b745153-4820-41b9-9ff8-2488dcb81e46.png)
what

**Comment by steviegt6** [comment]
> 
> 
> ![image](https://user-images.githubusercontent.com/75984453/126480818-5b745153-4820-41b9-9ff8-2488dcb81e46.png)
> what

Someone just referenced this issue on accident when merging a PR from some other repository. GitHub can be finicky wit that,
